nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/PutS3Object.java

```diff
@@ -458,6 +471,13 @@ public void process(final InputStream rawIn) throws IOException {
                             // single part upload
                             //----------------------------------------
                             final PutObjectRequest request = new PutObjectRequest(bucket, key, in, objectMetadata);
+                            if (keyId != null) {
+                                    if (!context.getProperty(SIGNER_OVERRIDE).getValue().equals("AWSS3V4Signer")) {
```

Comment:
Would it be enough to check that it's not V2?  I don't think we need to make it impossible to get wrong, as long as we make a good faith attempt to help them get it right.  I'm thinking of a few things - 

1. The default should now be V4, I would prefer we not force users to nail down their signature version
1. AWS regions and SDK versions are complicated, for example we don't check if your region supports V4
1. If or when AWS comes out with signature V5, we would have to update this field